File: 035.rb
# 035
# Search Insert Position
# 2015-05-11
#####################################################
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search_insert(nums, target)
  left = 0
  right = nums.size - 1
  while left < right
    mid = (left + right) / 2
    target <= nums[mid] ? right = mid : left = mid+1
  end
  target > nums[left] ? left + 1 : left
end

#####################################################

nums = [1, 3, 5, 6]
target = 0

puts search_insert(nums, target)

File: leetcode/leetcode/035.cpp
//
//  035.cpp
//  leetcode
//
//  Created by Hang Zhang on 5/11/15.
//  Copyright (c) 2015 Hilings Studio. All rights reserved.
//

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0, right = (int)nums.size() - 1;
        while (left < right) {
            int mid = (left + right) / 2;
            target <= nums[mid] ? right = mid : left = mid+1;
        }
        return target > nums[left] ? left + 1 : left;
    }
};

int main(int arg, char *argv[]) {
    // insert code here...
    cout << "LeetCode 035 Search Insert Position, C++ ...\n\n";

    int a[] = {1, 3, 5, 6};
    int target = 8;
    vector<int> nums (a, a+sizeof(a)/sizeof(int));

    Solution sol;
    cout << sol.searchInsert(nums, target);

    return 0;
}